alma/tests/Unit/Factories/ModuleFactoryTest.php

```diff
@@ -35,23 +37,83 @@ class ModuleFactoryTest extends TestCase
      */
     protected $moduleFactory;
 
+    /**
+     * @var ModuleFactoryBuilder
+     */
+    protected $moduleFactoryBuilder;
+
+    /**
+     * @var \Mockery\Mock|(\Mockery\MockInterface&ToolsHelper)
+     */
+    protected $toolsHelperMock;
+
     public function setUp()
     {
-        $this->moduleFactory = new ModuleFactory();
+        $this->moduleFactoryBuilder = new ModuleFactoryBuilder();
+        $this->moduleFactory = $this->moduleFactoryBuilder->getInstance();
+        $this->toolsHelperMock = \Mockery::mock(ToolsHelper::class)->makePartial();
+    }
+
+    public function tearDown()
+    {
+        $this->moduleFactory = null;
+        $this->moduleFactoryBuilder = null;
+        $this->toolsHelperMock = null;
     }
 
-    public function testGetContext()
+    public function testGetModule()
     {
         $this->assertInstanceOf(\Module::class, $this->moduleFactory->getModule());
     }
 
-    public function testGetContextName()
+    public function testGetModuleNameNoModule()
+    {
+        $moduleFactoryMock = \Mockery::mock(ModuleFactory::class, [$this->toolsHelperMock])->makePartial();
+        $moduleFactoryMock->shouldReceive('getModule')->andReturn(false);
+        $this->assertEquals('', $moduleFactoryMock->getModule());
+
+        $this->assertEquals(ConstantsHelper::ALMA_MODULE_NAME, $this->moduleFactory->getModuleName());
+    }
+
+    public function testGetModuleName()
     {
+        $moduleFactoryMock = \Mockery::mock(ModuleFactory::class, [$this->toolsHelperMock])->makePartial();
+
+        $moduleMock = \Mockery::mock(\Module::class)->makePartial();
+        $moduleMock->name = ConstantsHelper::ALMA_MODULE_NAME;
+        $moduleFactoryMock->shouldReceive('getModule')->andReturn($moduleMock);
+
         $this->assertEquals(ConstantsHelper::ALMA_MODULE_NAME, $this->moduleFactory->getModuleName());
     }
 
     public function testL()
     {
         $this->assertEquals('Pay now by credit card', $this->moduleFactory->l('Pay now by credit card', ConstantsHelper::SOURCE_CUSTOM_FIELDS));
     }
+
+    public function testIsInstalledPsAfter17()
+    {
+        $this->toolsHelperMock->shouldReceive('psVersionCompare')->andReturn(false);
+
+        $moduleFactory = \Mockery::spy(ModuleFactory::class, [$this->toolsHelperMock])->makePartial();
+        $moduleFactory->shouldReceive('isInstalledAfter17');
+
+        $moduleFactory->isInstalled('alma');
+
+        $moduleFactory->shouldHaveReceived('isInstalledAfter17');
+        $moduleFactory->shouldNotHaveReceived('isInstalledBefore17');
+    }
+
+    public function testIsInstalledPsBefore17()
+    {
+        $this->toolsHelperMock->shouldReceive('psVersionCompare')->andReturn(true);
+
+        $moduleFactory = \Mockery::spy(ModuleFactory::class, [$this->toolsHelperMock])->makePartial();
+        $moduleFactory->shouldReceive('isInstalledBefore17');
```

Comment:
not necessary

Reply:
without makePartial, the test doesn't work :/

Reply:
Ping me when you change - a try and that work

Reply:
We'll see together in synch?